### ISSUE TITLE
Cannot Set Breakpoint with Absolute Path

### ISSUE THREAD

**Issue by StandoffVenus** [go-delve/delve#2992]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
> Delve Debugger
Version: 1.8.2
Build: $Id: dbb493ec14d1e7753504d016b1e1ef1665b75b16 $

2. What version of Go are you using? (`go version`)?
> go1.17.7 darwin/arm64 

3. What operating system and processor architecture are you using?
> Darwin/ARM64 (OSX on Apple M1)

4. What did you do?
```
dlv debug cmd/duckdns/main.go
dlv> b /Users/mule/develop/code/duck-it/cmd/duckdns/main.go:29
```

5. What did you expect to see?
```
Breakpoint 1 set at 0x1023fc4c4 for main.main()
```

6. What did you see instead?
```
Command failed: location "/Users/mule/develop/code/duck-it/cmd/duckdns/main.go:29" not found
```

FWIW, I've found that using `b main.go:29` does work. No idea why the absolute path is failing.

**Comment by aarzilli** [comment]
> FWIW, I've found that using `b main.go:29` does work. No idea why the absolute path is failing.

What's the output of `b main.go:29`?

**Comment by StandoffVenus** [comment]
```
(dlv) b main.go:29
Breakpoint 1 set at 0x1003e44c4 for main.main() main.go:29
```

**Comment by aarzilli** [comment]
Can you redo this after passing `--log --log-output=rpc` to `dlv` and post the output? Both for `b main.go:29` and the one with the full path.

**Comment by aarzilli** [comment]
PS. The output you posted doesn't make much sense to me. If you ran `dlv debug ...` then the executable would contain full source paths and the output of `b main.go:29` would end in `./main.go:29` (or with a full file path, depending on the cwd) not `main.go:29` which strongly suggests that `-trimpath` was passed somewhere.
Either you have `-trimpath` in $GOFLAGS or something doesn't add up.

**Comment by StandoffVenus** [comment]
Noticed you mentioned `GOFLAGS` and `-trimpath`; I'm using Nix to manage my system/development and I'm using the `goBuildModule` derivation to build my Go projects. The derivation sets `GOFLAGS=-mod=vendor -trimpath`.

I took the liberty of setting `GOFLAGS=-mod=vendor`, then executing Delve again. The `b /Users/...` command executes successfully!

I did run `dlv debug` with the flags you provided - output is huge, so I didn't include it here. If you're still interested in it, I can attach it to another comment.
